core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -170,17 +170,29 @@ var ExplorationPlayerPage = function() {
   };
 
   this.submitFeedback = function(feedback) {
-    feedbackPopupLink.click();
+    feedbackPopupLink.isDisplayed().then(function(){
```

Comment:
Not totally sure what's going on here. Why the addition of isDisplayed() -- and what if it's not displayed? Shouldn't we error?

Ditto elsewhere, below.

Reply:
OK so previously the code used to look like

`feedbackPopupLink.click();`

I added a promise to it.

`feedbackPopupLink.then(function()({`
   `feedbackPopupLink.click();`
`});`

Upon seeing [isDisplayed](https://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.isDisplayed) docs and verifying it against [isPresent](https://stackoverflow.com/a/28120175), I proceeded to put it into clicking action whenever Travis test fails at that particular clicking action. 

With these justification, do you think `isDisplayed` is overkill? After all, it is a mean to getting clicks done in a multitude of [other means](https://stackoverflow.com/questions/36871071/click-function-isnt-working-in-protractor-scripts) With regards to error-ing out, I left it to protractor:

 `Failed: No element found using locator: By.css(.protractor-test-exploration-feedback-popup-link)`

for further investigation. Would throwing a `Error('feedbackPopupLink is not clickable')` be more helpful?

Reply:
I agree with using isDisplayed() for expectations. I have changemy isDisplayed() usage to    
    `expect(component.isDisplayed()).toBe(true);`
    `component.click()`